src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHostConfiguration.cs

```diff
@@ -94,11 +94,22 @@ public class WebAssemblyHostConfiguration : IConfiguration, IConfigurationRoot,
         /// <returns>The configuration sub-sections.</returns>
         IEnumerable<IConfigurationSection> IConfiguration.GetChildren()
         {
-            return _providers
-                .SelectMany(s => s.GetChildKeys(Enumerable.Empty<string>(), null))
-                .Distinct(StringComparer.OrdinalIgnoreCase)
-                .Select(key => this.GetSection(key))
-                .ToList();
+            var hashSet = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Completely unrelated question for my own insight, why does WASM need to be able to build/modify configuration providers like this?  (Deviating from the configure/build/immutable configuration pattern)

Reply:
The host builder in WASM is a single shot operation similar to the feather http builder. You configure services on it and tell it to run. We found that you frequently need access to configuration as you're building it - hence it has this read / write behavior. I thought there was an issue filed by @davidfowl to add support for this by default in config, but I can't seem to find it.

Reply:
@halter73 is gonna file it soon.